server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -83,9 +83,24 @@ public class RemoteClusterStateService implements Closeable {
 
     private static final Logger logger = LogManager.getLogger(RemoteClusterStateService.class);
 
-    // TODO make this two variable as dynamic setting [issue: #10688]
-    public static final int INDEX_METADATA_UPLOAD_WAIT_MILLIS = 20000;
-    public static final int GLOBAL_METADATA_UPLOAD_WAIT_MILLIS = 20000;
+    // default value for index metadata upload wait is 20s
+    static volatile TimeValue indexMetadataUploadWaitTime = TimeValue.timeValueMillis(20000);
```

Comment:
Do they need to be static?

Reply:
I thought it makes more sense to keep this static since having a common value across different objects made sense to me here, rather than allowing each object to have different value. 

Any reason to not keep it static?

Reply:
the variables are static but we would have multiple addSettingsUpdateConsumer for each object updating this value. i think it would be better to have to object level then.

Reply:
yeah makes sense, the class doesn't have any singleton class restrictions to ensure only 1 object exists at all time.

so will go ahead and make this change

Reply:
made the change to keep it private and non-static.

added a static default value that is consumed in the test class